packages/astro/src/internal/fetch-content.ts

```diff
@@ -11,12 +11,10 @@ export function fetchContent(importMetaGlobResult: Record<string, any>, url: str
         return;
       }
       const urlSpec = new URL(spec, url).pathname.replace(/[\\/\\\\]/, '/');
-      if (!urlSpec.includes('/pages/')) {
-        return mod.__content;
-      }
       return {
         ...mod.__content,
-        url: urlSpec.replace(/^.*\/pages\//, '/').replace(/\.md$/, ''),
+        url: urlSpec.includes('/pages/') && urlSpec.replace(/^.*\/pages\//, '/').replace(/\.md$/, ''),
```

Comment:
Should this be updated to use `astroConfig.pages` (if a user has changed that path)?

Reply:
I'm a little nervous to change this as a part of this PR since I don't really know this code well. Okay to leave for a future PR? It may even be good to finalize our `pages` support since I think at one point we wanted to hardcode `pages` to `src/pages`, and this code may be left over from that.